nativelib/src/main/resources/gc/immix/metadata/BlockMeta.h

```diff
@@ -9,45 +11,90 @@
 
 typedef enum {
     block_free = 0x0,
-    block_recyclable = 0x1,
-    block_unavailable = 0x2
+    block_simple = 0x1,
+    block_superblock_start = 0x2,
+    block_superblock_middle = 0x3,
+    block_marked = 0x5
 } BlockFlag;
 
 typedef struct {
-    uint8_t mark;
-    uint8_t flags;
-    int16_t first;
+    union {
+        struct {
+            uint8_t flags;
+            int8_t first;
+        } simple;
+        struct {
+            uint8_t flags;
+            int32_t size : BLOCK_COUNT_BITS;
+        } superblock;
+    } block;
```

Comment:
BlockMeta struct stays 64-bit sized in the end, which is great. https://godbolt.org/z/9GXIan